src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,63 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+
+    struct TrafficGauges{
+        TrafficGauges (char const* name, beast::insight::Collector::ptr const& collector)
+        : bytesIn(collector->make_gauge(name,"Bytes_In"))
+        , bytesOut(collector->make_gauge(name,"Bytes_Out"))
+        , messagesIn(collector->make_gauge(name,"Messages_In"))
+        , messagesOut(collector->make_gauge(name,"Messages_Out"))
+        {   
+        }
+        beast::insight::Gauge bytesIn;
+        beast::insight::Gauge bytesOut;
+        beast::insight::Gauge messagesIn;
+        beast::insight::Gauge messagesOut;
+    };
+
+    
+    struct Stats
+    {
+
+        template <class Handler>
+        Stats (
+                Handler const& handler, 
+                beast::insight::Collector::ptr const& collector,
+                std::vector<TrafficGauges>&& trafficGauges_)
+            : peerDisconnects (collector->make_gauge("Overlay","Peer_Disconnects"))
+            , trafficGauges (std::move(trafficGauges_))
+            , hook (collector->make_hook (handler))
+            { 
+            }
+            
+        beast::insight::Gauge peerDisconnects;
+        std::vector<TrafficGauges> trafficGauges;
+        beast::insight::Hook hook;
+    };
+    
+    Stats m_stats;
+    std::mutex m_statsMutex;
+
+private:
+    void collect_metrics()
+    {   
+        std::lock_guard lock (m_statsMutex);
+        auto counts = m_traffic.getCounts();
+
+        for (std::size_t i = 0; i < counts.size(); ++i)
+        {
+            assert(counts.size() == m_stats.trafficGauges.size());
```

Comment:
nit: can move this assert to a few lines up, outside of the loop (it only needs to be checked once)

Reply:
Thanks, I fixed it.